### ISSUE TITLE
[DOC] Request for an update in the iloc documentation

### ISSUE THREAD

**Issue by Salonijain27** [rapidsai/cudf#4200]
The iloc in cudf documentation for 0.12 and 0.13 shows:
```
>>> df = DataFrame([('a', list(range(20))),
...                 ('b', list(range(20))),
...                 ('c', list(range(20)))])
```
https://rapidsai.github.io/projects/cudf/en/0.12.0/api.html#cudf.core.dataframe.DataFrame.iloc
https://rapidsai.github.io/projects/cudf/en/0.13.0/api.html#cudf.core.dataframe.DataFrame.iloc
which I believe is no longer the correct way of creating a cudf dataframe.
Using the above mentioned code throws the following error:
```
In [1]: from cudf import DataFrame
   ...: df = DataFrame([('a', list(range(20))),
   ...:                 ('b', list(range(20))),
   ...:                 ('c', list(range(20)))])
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-1-5a8029207caf> in <module>
      2 df = DataFrame([('a', list(range(20))),
      3                 ('b', list(range(20))),
----> 4                 ('c', list(range(20)))])
~/miniconda3/envs/check-mnmg/lib/python3.7/site-packages/cudf/core/dataframe.py in __init__(self, data, index, columns, dtype)
    193                 if len(data) > 0 and is_scalar(data[0]):
    194                     data = [data]
--> 195                 self._init_from_list_like(data, index=index, columns=columns)
    196
    197             else:
~/miniconda3/envs/check-mnmg/lib/python3.7/site-packages/cudf/core/dataframe.py in _init_from_list_like(self, data, index, columns)
    219
    220         for col_name, col in enumerate(data):
--> 221             self._data[col_name] = column.as_column(col)
    222
    223         if not hasattr(self, "multi_cols"):
~/miniconda3/envs/check-mnmg/lib/python3.7/site-packages/cudf/utils/utils.py in __setitem__(self, key, value)
    347             if len(value) != len(self.first):
    348                 raise ValueError(
--> 349                     f"Cannot insert Column of different length "
    350                     "into OrderedColumnDict"
    351                 )
ValueError: Cannot insert Column of different length into OrderedColumnDict
```

**Comment by shwina** [comment]
Thanks for reporting. We should instead have:

```
DataFrame({'a': range(20), 'b': range(20), 'c': range(20)})
```

**Comment by rnyak** [comment]
@shwina I can update the docs.

**Comment by wbadart** [comment]
For any other lurkers tracking this issue, looks like this is fixed by 52721ba84b

**Comment by kkraus14** [comment]
Fixed by #4825
